politeiad/backendv2/tstorebe/plugins/comments/cmds.go

```diff
@@ -0,0 +1,1532 @@
+			util.Digest(b), digest)
+	}
+	var c comments.CommentAdd
+	err = json.Unmarshal(b, &c)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal CommentAdd: %v", err)
+	}
+
+	return &c, nil
+}
+
+func convertCommentDelFromBlobEntry(be store.BlobEntry) (*comments.CommentDel, error) {
+	// Decode and validate data hint
+	b, err := base64.StdEncoding.DecodeString(be.DataHint)
+	if err != nil {
+		return nil, fmt.Errorf("decode DataHint: %v", err)
+	}
+	var dd store.DataDescriptor
+	err = json.Unmarshal(b, &dd)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal DataHint: %v", err)
+	}
+	if dd.Descriptor != dataDescriptorCommentDel {
+		return nil, fmt.Errorf("unexpected data descriptor: got %v, want %v",
+			dd.Descriptor, dataDescriptorCommentDel)
+	}
+
+	// Decode data
+	b, err = base64.StdEncoding.DecodeString(be.Data)
+	if err != nil {
+		return nil, fmt.Errorf("decode Data: %v", err)
+	}
+	digest, err := hex.DecodeString(be.Digest)
+	if err != nil {
+		return nil, fmt.Errorf("decode digest: %v", err)
+	}
+	if !bytes.Equal(util.Digest(b), digest) {
+		return nil, fmt.Errorf("data is not coherent; got %x, want %x",
+			util.Digest(b), digest)
+	}
+	var c comments.CommentDel
+	err = json.Unmarshal(b, &c)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal CommentDel: %v", err)
+	}
+
+	return &c, nil
+}
+
+func convertCommentVoteFromBlobEntry(be store.BlobEntry) (*comments.CommentVote, error) {
+	// Decode and validate data hint
+	b, err := base64.StdEncoding.DecodeString(be.DataHint)
+	if err != nil {
+		return nil, fmt.Errorf("decode DataHint: %v", err)
+	}
+	var dd store.DataDescriptor
+	err = json.Unmarshal(b, &dd)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal DataHint: %v", err)
+	}
+	if dd.Descriptor != dataDescriptorCommentVote {
+		return nil, fmt.Errorf("unexpected data descriptor: got %v, want %v",
+			dd.Descriptor, dataDescriptorCommentVote)
+	}
+
+	// Decode data
+	b, err = base64.StdEncoding.DecodeString(be.Data)
+	if err != nil {
+		return nil, fmt.Errorf("decode Data: %v", err)
+	}
+	digest, err := hex.DecodeString(be.Digest)
+	if err != nil {
+		return nil, fmt.Errorf("decode digest: %v", err)
+	}
+	if !bytes.Equal(util.Digest(b), digest) {
+		return nil, fmt.Errorf("data is not coherent; got %x, want %x",
+			util.Digest(b), digest)
+	}
+	var cv comments.CommentVote
+	err = json.Unmarshal(b, &cv)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal CommentVote: %v", err)
+	}
+
+	return &cv, nil
+}
+
+func convertCommentFromCommentAdd(ca comments.CommentAdd) comments.Comment {
+	return comments.Comment{
+		UserID:        ca.UserID,
+		Token:         ca.Token,
+		ParentID:      ca.ParentID,
+		Comment:       ca.Comment,
+		PublicKey:     ca.PublicKey,
+		Signature:     ca.Signature,
+		CommentID:     ca.CommentID,
+		Version:       ca.Version,
+		Timestamp:     ca.Timestamp,
+		Receipt:       ca.Receipt,
+		Downvotes:     0, // Not part of commentAdd data
```

Comment:
That's what the `Comment` structure is.  Its a wrapper around the `CommentAdd`, `CommentDel`, and `CommentVote` structures, which are the structures that are saved to disk.  Its just flattened out because the client doesn't need to know about the internals of how comments are saved and updated in the backend.